components/src/core/components/SchemaForm/SchemaForm.vue

```diff
@@ -178,7 +178,9 @@ export default defineComponent({
                 required:
                   field.validators?.has('required') && !props.schema?.disabled,
                 ...(field.type !== 'custom' && {type: field.type}),
-              }),
+                setDirty: field.setDirty,
```

Comment:
Can't we pass the dirty prop as prop of schema form field rather than using a new field.

Reply:
fixed